Update IPC serialization for LayerTreeContext

sheeparegreat opened the pull request:
#### d04fd46f576740058cb078ae40c6fad00faf3d16
<pre>
Update IPC serialization for LayerTreeContext
<a href="https://bugs.webkit.org/show_bug.cgi?id=245251">https://bugs.webkit.org/show_bug.cgi?id=245251</a>
rdar://100000954

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/LayerTreeContext.cpp:
(WebKit::LayerTreeContext::encode const): Deleted.
(WebKit::LayerTreeContext::decode): Deleted.
* Source/WebKit/Shared/LayerTreeContext.h:
(WebKit::LayerTreeContext::LayerTreeContext):
* Source/WebKit/Shared/LayerTreeContext.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Continue porting old argument coders over to the new generated
serialized coders.

Canonical link: <a href="https://commits.webkit.org/254694@main">https://commits.webkit.org/254694@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36eacf60b81aded9b4facbbfb96a690606a09425

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89478 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34032 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20215 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98795 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32535 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28021 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81836 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93208 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95125 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25845 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76373 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25777 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80723 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80621 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68766 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30294 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14695 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30034 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15632 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3296 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33481 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38588 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32191 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34723 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/LayerTreeContext.h:
```diff
@@ -45,11 +40,9 @@ enum class LayerHostingMode : uint8_t {
 class LayerTreeContext {
 public:
     LayerTreeContext();
```
This constructor now has uninitialized memory.  Can it be removed?  If both are removed, it should just compile.

sheeparegreat committed:
Update IPC serialization for LayerTreeContext
https://bugs.webkit.org/show_bug.cgi?id=245251
rdar://100000954

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/LayerTreeContext.cpp:
(WebKit::LayerTreeContext::encode const): Deleted.
(WebKit::LayerTreeContext::decode): Deleted.
* Source/WebKit/Shared/LayerTreeContext.h:
(WebKit::LayerTreeContext::LayerTreeContext):
* Source/WebKit/Shared/LayerTreeContext.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Continue porting old argument coders over to the new generated
serialized coders.

Canonical link: https://commits.webkit.org/254694@main
